### PR TITLE
Introduce a `zip` function for merging lists

### PR DESCRIPTION
**Pull request by dominiklohmann**
This implements `zip` with the following signature:

```
zip(left: list<T>, right: list<U>) -> list<record{left: T, right: U}>
```

### CODE REVIEW

File: libtenzir/builtins/functions/list.cpp [hunk 1]
```diff
@@ -116,6 +118,81 @@ class concatenate : public virtual function_plugin {
...
+      for (auto i = int64_t{0}; i < left.length(); ++i) {
+        auto left_value = check(left_values.next());
+        auto right_value = check(right_values.next());
+        if (not left_value and not right_value) {
+          builder.null();
+          continue;
+        }
+        auto list_builder = builder.list();
+        const auto max_length = std::max(left_value ? left_value->size() : 0,
+                                         right_value ? right_value->size() : 0);
```
Comment (dominiklohmann) [inline libtenzir/builtins/functions/list.cpp:178]: @jachris brought up that perhaps by default we should warn when the list lengths do not match, introducing an option to choose whether we prefer the min or the max length or warning and returning null.

Comment (dominiklohmann) [reply libtenzir/builtins/functions/list.cpp:178]: We decided internally to always warn if the lengths of the lists do not match or if either (but not both) list are null. We want to use the max length here in that case.

We can always add options in the future for toggling this behavior, but for now we'll stick with this without options.

File: web/docs/tql2/functions/zip.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+## Description
+
+The `zip` function returns a list containing records with two fields `left` and
+`right`, each containing the respective elements of the input lists.
+
+If both lists are null, `zip` returns null. If one of the lists is null or has a
+mismatching length, missing values are filled in with nulls, using the longer
+list's length and emitting a warning.
```
Comment (jachris) [inline web/docs/tql2/functions/zip.md:16]: Some comma changes & "emitting a warning"
```suggestion
If both lists are null, `zip` returns null. If one of the lists is null or has a
mismatching length, missing values are filled in with nulls using the longer
list's length, and a warning is emitted.
```

**Review by jachris — APPROVED** [review]
Some tests would be nice if we have the time to add them now.

File: libtenzir/builtins/functions/list.cpp
Comment (jachris) [inline libtenzir/builtins/functions/list.cpp:1]: Some tests would be nice if we have the time to add them now.

**Pull request merged**